sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/QueryPlanningTracker.scala

```diff
@@ -89,8 +91,35 @@ object QueryPlanningTracker {
   }
 }
 
+/**
+ * Callbacks after planning phase completion.
+ *
+ * @since 3.5.0
+ */
+@DeveloperApi
+private[sql] abstract class QueryPlanningTrackerCallback {
```

Comment:
This one you won't have to make it explicitly `private[sql]` as we already documented this whole package is private (https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/package.scala#L21-L22).